### DISCUSSION TITLE
Apollo Community Meeting 2024/04/11 08:00PM UTC+8

### DISCUSSION THREAD

**Discussion by hezhangjian** [apolloconfig/apollo#5119] [General]
Hello everyone, we're going to have a community meeting at 2024/04/11 08:00PM UTC+8. We will talk about:
- 2024 开源之夏活动
- everything else you want to talk.

Attention：We will use chinese instead of english in the meeting

会议主题：Apollo社区会议
会议时间：2024/04/11 20:00-21:30 (GMT+08:00) 中国标准时间 - 北京

点击链接入会，或添加至会议列表：
https://meeting.tencent.com/dm/g7MnOMYlia9R

#腾讯会议：681-919-466

**Comment by nobodyiam** [comment]
+1

**Comment by weetyre** [comment]
+1

**Comment by hezhangjian** [comment]
ApolloConfig会议纪要:

与会人：宋顺(nobodyiam)、小权(Anilople)、邹毅贤(zouyx)、张乐(lepdou)、陈凯玲、天行、blackbear2003、张俭(shoothzj)、王乾

# 议题一：OSPP项目讨论（张俭、小权、张乐）

## Java、Golang客户端缓存支持Kubernetes ConfigMap

- 客户端缓存支持SPI接口
- Kubernetes ConfigMap的缓存作为插件方式提供
- 需要考虑kubernetes的性能

注意点：

- configmap和s3的api可能是整读整写，设计configmap或s3文件的粒度，避免数据冲突覆盖


## 为 Apollo 管理员工具添加全局搜 Value 的功能

## 为apollo-client添加更多可观测性

# 议题二：开放讨论（ALL）

## 易用性增强

- 公共namespace配置中，考虑添加提示，说明有哪些应用覆盖了配置
- 现在ApolloConfig已经支持3个jar包合一部署，并且支持内存数据库h2，将demo.sh拆分成许多可执行的start脚本，更方便用户使用

## 可靠性增强

- apollo-client添加应急场景下，下发配置的能力。在apollo-configservice及portal都宕机时，可以修改配置。

## 功能扩展

- apollo-portal方便扩展和二次开发的能力。增强openapi，或者将apollo-portal变成可引入的二方库，用户可在工程中使用注解@EnableApolloPortal后，进行二次开发，参考Eureka的@EnableEurekaServer

**Comment by hezhangjian** [comment]
与会人：邹毅贤、王乾、刁一轩、贺张俭

1. 需要定义出接口的定义，并在社区评审通过，加上顺哥，小权哥一起
2. java和go的有差异，需要注意行为
3. 需要在代码中进行测试，需要了解github上如何运行kubernetes
4. configmap没必要走那么快，先定义接口，先将原本的文件缓存重构成接口的一种实现
5. 熟悉go的话，可以先做一个demo，做一个登陆的动作，通过apollo在控制某个用户到生产还是到灰度
